packages/compartment-mapper/src/assemble.js

```diff
@@ -2,15 +2,110 @@
 /// <reference types="ses" />
 
 import { resolve } from './node-module-specifier.js';
-import { mapParsers } from './parse.js';
+import { parseExtension } from './extension.js';
 
-const { entries, freeze } = Object;
+const { entries, fromEntries, freeze } = Object;
 
 const defaultCompartment = Compartment;
 
 // q, as in quote, for strings in error messages.
 const q = JSON.stringify;
 
+/**
+ * @template {string | number} Key
+ * @template Value
+ * @param {Record<Key, Value>} object
+ * @param {Key} key
+ * @returns {boolean}
+ */
+const has = (object, key) => Object.prototype.hasOwnProperty.call(object, key);
```

Comment:
Assuming we're trying to be correct without SES

```suggestion
const { apply } = Reflect;
const { hasOwnProperty } = Object.prototype;
const has = (object, key) => apply(hasOwnProperty, object, [key]);
```
or anything you like that's as robust.